engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -262,28 +230,49 @@ case List(expr) => Some(expr)
         var translatedConstructors = CollectionConverters.asScala(translatedConstructorsIt).toList();
         // type
         List<IR.DefinitionArgument> args = translateArgumentsDefinition(def.getParams());
-        yield new IR$Module$Scope$Definition$SugaredType(
+        var t = new IR$Module$Scope$Definition$SugaredType(
           typeName,
           args,
           translatedConstructors.appendedAll(translatedBody),
           getIdentifiedLocation(inputAst),
           meta(), diag()
         );
+        yield cons(t, appendTo);
       }
       case Tree.Function fn -> {
         var methodRef = translateMethodReference(fn.getName(), false);
         var args = translateArgumentsDefinition(fn.getArgs());
         var body = translateExpression(fn.getBody(), false);
 
-        yield new IR$Module$Scope$Definition$Method$Binding(
+        var b = new IR$Module$Scope$Definition$Method$Binding(
           methodRef,
           args,
           body,
           getIdentifiedLocation(inputAst),
           meta(), diag()
         );
+        yield cons(b, appendTo);
       }
-      /*
+      case Tree.Annotated anno -> {
+        var n = new IR$Name$Annotation("@" + anno.getAnnotation().codeRepr(), getIdentifiedLocation(anno), meta(), diag());
+        yield translateModuleSymbol(anno.getExpression(), cons(n, appendTo));
+      }
+      case Tree.Documented doc -> {
+        var c = translateComment(doc, doc.getDocumentation());
```

Comment:
`c` as in a comment?

Reply:
`c` as a comment. `t` as a tree. `b` as a block, `n` as annotation.